File: CWM/Models/UserInsertUpdate.cs
using CWM.Core.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace CWM.Models
{
    public class UserInsertUpdate
    {
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public Gender Gender { get; set; }
        public DateTime? BirthDate { get; set; }
        public DateTime? CreateDate { get; set; }
        public byte[]? Image { get; set; }
        public string Email { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public string Mobile { get; set; } = string.Empty;
        public string OfficePhone { get; set; } = string.Empty;
        public List<Role> Roles { get; set; } = new();

        [Required]
        public int? CityId { get; set; }

        [Required]
        public int? CountryId { get; set; }
    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/requests/user_insert_update.dart
import 'package:cwm_desktop_mobile/models/enums/gender.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_insert_update.g.dart';

@JsonSerializable()
class UserInsertUpdate {
  String firstName;
  String lastName;
  Gender gender;
  DateTime birthDate;
  String? cityId;
  String? citizenshipId;
  String image;
  String username;
  String password;
  String email;
  String mobile;

  UserInsertUpdate(
    this.firstName,
    this.lastName,
    this.gender,
    this.birthDate,
    this.cityId,
    this.citizenshipId,
    this.image,
    this.username,
    this.password,
    this.email,
    this.mobile,
  );

  factory UserInsertUpdate.fromJson(Map<String, dynamic> json) =>
      _$UserInsertUpdateFromJson(json);

  Map<String, dynamic> toJson() => _$UserInsertUpdateToJson(this);
}
